### PR TITLE
[SCB-1389] write docs for explicit passing transaction context

### PR DESCRIPTION
**Pull request by chanjarster**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: docs/user_guide.md [hunk 1]
```diff
@@ -111,7 +111,108 @@ Take a transfer money application as an example:
...
+}
+@Compensable
+public void bar(BarCommand cmd, TransactionContext parentTxContext) {
+  ...
+  // TransactionContext childTxContext = omegaContext.getTransactionContext();
+}
+```
+
+Notice that `bar` method got parent transaction context in parameter list, and got child transaction context from `OmegaContext` in method body. So if you want to passing transaction context to another service, you should pass child transaction context.
+
```
Comment (zhfeng) [inline docs/user_guide.md]: The pack does not support the nested transaction right now. see [SCB-836](https://issues.apache.org/jira/browse/SCB-836) and I raised the https://github.com/apache/servicecomb-pack/pull/432 but it did not finish. So I think we could add a note here and reference to the SCB-836.

Comment (chanjarster) [reply docs/user_guide.md]: @zhfeng  I think what I mentioned here is nested local tx not nested saga though

Comment (zhfeng) [reply docs/user_guide.md]: well, it could be better to use the injected or imported transaction context here.

Comment (zhfeng) [reply docs/user_guide.md]: parent transaction context -> injected (imported) transaction context
child transaction context -> local (exported) transaction context

Comment (zhfeng) [reply docs/user_guide.md]: Anyway, if the users want to use the chained service invoking which means service A -> B -> C, it looks like they would like to pass the explicit transaction context from the parameter list when B -> C.

**Comment by zhfeng** [comment]
Thanks @chanjarster - it looks good to me.

File: docs/user_guide.md [hunk 2]
```diff
@@ -111,7 +111,108 @@ Take a transfer money application as an example:
...
+@Compensable
+public void bar() {
+  ...
+}
+```
+
+Here is how Omega does:
+
+1. Service A's `foo` method opens a new global transaction.
+2. `TransactionClientHttpRequestInterceptor` injects transaction context into request headers when `RestTemplate` request Service B.
```
Comment (WillemJiang) [inline docs/user_guide.md]: Can we add the file link for the interceptor?

File: docs/user_guide.md [hunk 3]
```diff
@@ -111,7 +111,108 @@ Take a transfer money application as an example:
...
+3. When Service B receive the request, `TransactionHandlerInterceptor` extract context info from request headers.
+
+Omega supports following implicity transaction context passing:
+
+1. omega-transport-{dubbo,feign,resttemplate,servicecomb}.
+2. Method call in the same thread (based on `OmegaContext` thread local fields).
+3. `java.util.concurrent.Executor{Service}` annotated by `@OmegaContextAware`.
+
+So here comes a problem: what if implicit transaction context passing can't work? For example, Service A invokes Service B via some RPC library and no extension can be made to injecting or extracting transaction context information. In this situation you need explicit transaction context passing. Omega provides two classes to achieve that.
```
Comment (WillemJiang) [inline docs/user_guide.md]: Please specify the pack version we provide here.

File: docs/user_guide.md (hunk 2 above)
Comment (chanjarster) [reply docs/user_guide.md]: Resolved

File: docs/user_guide.md (hunk 3 above)
Comment (chanjarster) [reply docs/user_guide.md]: Resolved

**Pull request merged**